### PR TITLE
Add supersourced files and RE2J.gwt.xml to the source jar instead of the binary jar

### PR DESCRIPTION
**Pull request by tjgq**
Tested by running `gradle jar` and verifying that the sources are included.

### CODE REVIEW

**Comment by tjgq** [comment]
N.B. I know next to nothing about Gradle. There might be a smarter way of writing this configuration.

**Comment by sjamesr** [comment]
There's already a re2j-sources jar published, if you can convince your
project's build system to add it to the gwt compiler classpath, you
shouldn't need this pull request

Come to think of it, that probably applies to PR #87 too :-)

**Comment by tjgq** [comment]
Depending on the source jar is indeed an option, but I still need to include the supersourced files and the `gwt.xml` file. I've updated this PR with that change. PTAL.

File: build.gradle [hunk 1]
```diff
@@ -58,6 +66,11 @@ sourceSets {
   }
 }
 
+jar {
+  // Include .java files for GWT compatibility.
+  from sourceSets.main.allJava
```
Comment (sjamesr) [inline build.gradle]: Is this typical for gwt-compatible projects? It seems wrong to include the sources in the binary jar, instead the gwt compiler should put the sources jar on its classpath.

Comment (tjgq) [reply build.gradle]: Sorry - what I actually meant to do in this PR was to add the xml+supersources into the sources jar, and leave the binary jar unchanged. Fixed.

**Pull request merged**